AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/AndroidKeyProviderTests.java

```diff
@@ -0,0 +1,173 @@
+package cloudant.com.androidtest.keyprovider;
+
+import android.test.AndroidTestCase;
+
+import com.cloudant.sync.datastore.encryption.AndroidKeyProvider;
+import com.cloudant.sync.datastore.encryption.DPKException;
+import com.cloudant.sync.datastore.encryption.EncryptionKey;
+import com.cloudant.sync.datastore.encryption.KeyManager;
+import com.cloudant.sync.datastore.encryption.KeyStorage;
+
+import java.util.Arrays;
+
+public class AndroidKeyProviderTests extends AndroidTestCase {
+
+    public void testCreateProviderWithIdentifier() {
+        AndroidKeyProvider provider = new AndroidKeyProvider(getContext(), "provider1password",
+                ProviderTestUtil.getUniqueIdentifier());
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
```

Comment:
I'm not sure this test is correct -- you are just testing the same key is returned by `getEncryptionKey` each time?